cmd/zoekt-sourcegraph-indexserver/fetcher.go

```diff
@@ -0,0 +1,167 @@
+package main
+
+import (
+	"bytes"
+	"context"
+	"fmt"
+	"github.com/google/zoekt"
+	"io"
+	"os/exec"
+	"strings"
+)
+
+// a deltaFetcher returns a list of changed files along with their file contents
+//
+// NOTE: The idea of returning the file contents in the same function seems awkward
+// with local checkouts, but necessary in a remote one
+type deltaFetcher interface {
+	// Fetch returns the paths that have changed on the latest
+	// commit for each branch, along with their file contents.
+	Fetch(ctx context.Context, branches ...zoekt.RepositoryBranch) (branchDeltas, error)
+}
+
+type branchDeltas map[string]delta
+
+// delta contains all the paths that have been added, changed, and removed
+// along with the file contents
+//
+// TODO@ggilmore: should the file contents  be io.ReadClosers, and should delta offer a "close"
+// method that closes all the file contents (thinking about the local file case)
+type delta struct {
+	// Added is a list of all the paths that have been added,
+	// along with their file contents.
+	Added map[string]io.Reader
+
+	// Changed is a list of all the paths that have been changed,
+	// along with their file contents.
+	Changed map[string]io.Reader
+
+	// Removed is a list of all the paths that were removed.
+	Removed []string
+}
+
+// NOTE: We can mimic the existing behavior (albeit with extra work on Zoekt's end)
+// by creating a fresh git checkout every time, feeding it to local fetcher,
+// and copying all the files into a new repository for zoekt-git-index to process. This
+// doesn't seem that useful, but we can do this to be able to merge a PR earlier.
```

Comment:
this sort of behaviour is useful for local integration tests though? So there is likely some justification for this.